server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -417,7 +459,7 @@ public void doRun()
               synchronized (hydrant) {
                 if (!hydrant.hasSwapped()) {
                   log.info("Hydrant[%s] hasn't swapped yet, swapping. Sink[%s]", hydrant, sink);
-                  final int rowCount = persistHydrant(hydrant, schema, interval);
+                  final int rowCount = persistHydrant(hydrant, schema, interval, null);
```

Comment:
If the implementation is not scoped to handle data dupe or HA then it is fully functional without those and the conversation on how to handle that expanded scope would be swallowed here.  Perhaps top comment needs updated to clarify scope. (smaller is better)